DataFormats/Detectors/CPV/include/DataFormatsCPV/RawFormats.h

```diff
@@ -62,29 +65,78 @@ class cpvword
   unsigned char bytes[16] = {0};
 };
 
+class cpvheader
+{
+ public:
+  cpvheader() = default;
+  cpvheader(std::vector<char>::const_iterator b, std::vector<char>::const_iterator e)
+  {                                               //reading header from file
+    for (int i = 0; i < 16 && b != e; i++, b++) { //read up to 16 bytes
+      bytes[i] = *b;
+    }
+  }
+  cpvheader(InteractionRecord orbitBC, bool isNoDataExpected, bool isDataContinued)
+  { //writing header
+    //header is 128-bit word.
+    //|127-120|119-112|111-104|103-96|95-88 |87-80 |79-72|71-64|63-56|55-48|47-40|39-32|31-24|23-16|15-8 |7-0  |
+    // byte15  byte14  byte13  byte12 byte11 byte10 byte9 byte8 byte7 byte6 byte5 byte4 byte3 byte2 byte1 byte0
+    //byte = |76543210|
+    bytes[0] = (0x010 & 0x0ff);                                               //bits 11 - 0 trigger id (0x010 = physics trigger)
+    bytes[1] = ((0x010 & 0xf00) >> 8)                                         //bits 11 - 0 trigger id (0x010 = physics trigger)
```

Comment:
(0x010 & 0xf00) isn't a typo?

Reply:
Dear @peressounko ,
no, this is not a typo. It's just "explained" zero. Trigger type TT is coded as first 12 bits. So bytes[0] = (TT & 0x0ff);    bytes[1] = ((TT & 0xf00) >> 8). I just hardcoded physical trigger in this formula so if we look at this later we will easily understand what is hardcoded there.